e2e/etcd_release_upgrade_test.go

```diff
@@ -86,3 +91,55 @@ func TestReleaseUpgrade(t *testing.T) {
 		}
 	}
 }
+
+func TestReleaseUpgradeWithSnapshot(t *testing.T) {
```

Comment:
This test case is not exactly what i described here: https://github.com/coreos/etcd/pull/6309#issuecomment-243523255

we need to restart the new cluster at the end of test.